src/plots/cartesian/autorange.js

```diff
@@ -174,15 +178,40 @@ function makePadFn(ax) {
     return function getPad(pt) { return pt.pad + (pt.extrapad ? extrappad : 0); };
 }
 
-function doAutoRange(ax) {
+function concatExtremes(gd, ax, k) {
```

Comment:
Hmm. Come to think of it, this thing will be a drag. Looping over all traces, annotations and shapes for every axis, doesn't scale very well.

Reply:
Improvements in -> 66df51e8cf9d99e69f3968eb5e73929c55c233a2